### PR TITLE
Add GitHub action to validate OpenAPI specs

### PR DESCRIPTION
**Pull request by bbrks**
Validates the OpenAPI specs when modified. Caught some existing failures, will leave those to be fixed separately.

- The configuration specified in `.redocly.yaml` determines what rules are run and which API specs exist, and is automatically found by the tools below:
  - GH Actions runs the Redocly OpenAPI lint tool whenever a file changes in `docs/api/**`
  - You can also run locally with `openapi lint` installable here: https://github.com/Redocly/openapi-cli#node

Screenshot 2022-03-18 at 16 44 39

## Dependencies
- [x] https://github.com/mhiew/redoc-lint-github-action/pull/2 

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- n/a

### CODE REVIEW

File: .redocly.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+apiDefinitions:
+  admin: ./docs/api/admin.yaml
+  public: ./docs/api/public.yaml
+  metric: ./docs/api/metric.yaml
+
+lint:
+  extends:
+    - minimal
+  rules:
```
Comment (IsaacLambat) [inline .redocly.yaml:19]: Would be good to allow examples to have additional properties that are not defined in the schema to stop some of the warnings (due to doc bodies having user defined properties): https://redocly.com/docs/cli/resources/built-in-rules/#no-invalid-schema-examples

**Review by IsaacLambat — CHANGES_REQUESTED** [review]

File: .redocly.yaml (hunk 1 above)
Comment (bbrks) [reply .redocly.yaml:19]: When I briefly looked a couple of weeks ago, I think I saw an in-spec way to describe additional user-defined fields in the schema, which seems like the more correct approach.

**Review by IsaacLambat — APPROVED** [review]
LGTM

File: .redocly.yaml (hunk 1 above)
Comment (bbrks) [reply .redocly.yaml:19]: IIRC it was using the `additionalProperties` field

**Pull request merged**